### PR TITLE
Update unit testing documentation

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

None

### Summary of the issue:

The current documentation for running unit tests is out of date, referring to tools that we no longer use, and commands that no longer work.

### Description of user facing changes

Updated the project documentation to reflect the status quo.

### Description of development approach

N/A

### Testing strategy:

Ran any command line examples to ensure they work as expected.

### Known issues with pull request:

None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **Documentation**
  - Enhanced unit testing instructions, detailing how to run specific tests using patterns with the `-k` option.
  - Updated documentation to reflect the use of `xmlrunner` for test execution, improving clarity on argument forwarding.

### CODE REVIEW

File: projectDocs/testing/automated.md [hunk 1]
```diff
@@ -39,10 +29,20 @@ To be warned about linting errors faster, you may wish to integrate Flake8 with
 For more details, see `tests/lint/readme.md`
 
 ### Unit Tests
+
 Unit tests can be run with the `rununittests.bat` script.
-Internally this script uses the Nose Python test framework to execute the tests.
-Any arguments given to `rununittests.bat` are forwarded onto Nose.
-Please refer to Nose's own documentation on how to filter tests etc.
+Internally this script uses the [xmlrunner](https://github.com/pycontribs/xmlrunner) wrapper around the [unittest](https://docs.python.org/3/library/unittest.html) framework to execute the tests.
+Any arguments given to `rununittests.bat` are forwarded onto xmlrunner, and thence to unittest.
```
Comment (seanbudd) [inline projectDocs/testing/automated.md:35]:
```suggestion
Any arguments given to `rununittests.bat` are forwarded onto xmlrunner, and then to unittest.
```

**Review by seanbudd — APPROVED** [review]

File: projectDocs/testing/automated.md
Comment (CyrilleB79) [inline projectDocs/testing/automated.md:1]: While at it, could you check line 5 that the list of tests that are run when running `scons test` is exhaustive?

**Review by Qchristensen — APPROVED** [review]
All looks fine to me from a readability point of view.
But I will leave it to others to determine that the commands etc are all correct.

Only comment is with the line "Please refer to xmlrunner's and unittest's own documentation for further information on how to filter tests etc." should we link to the documentation for xmlrunner and unittest?

File: projectDocs/testing/automated.md [hunk 2]
```diff
@@ -39,10 +22,20 @@ To be warned about linting errors faster, you may wish to integrate Ruff with ot
...
+
+To run only specific unit tests, specify a pattern to match against using the `-k` option on the command line.
+The `-k` option can be provided multiple times to provide multiple patterns to match against.
+For example, to run only methods in the `TestMove` and `TestSelection` classes in the file `tests\unit\test_cursorManager.py` file, run this command:
+
+```cmd
+rununittests -k test_cursorManager.TestMove -k test_cursorManager.TestSelection
+```
+
+Please refer to [unittest's documentation](https://docs.python.org/3/library/unittest.html#command-line-interface) for further information on how to filter tests etc.
```
Comment (seanbudd) [inline projectDocs/testing/automated.md:38]:
```suggestion
Please refer to [unittest's documentation](https://docs.python.org/3/library/unittest.html#command-line-interface) for further information on how to filter tests.
```

**Review by seanbudd — APPROVED** [review]

**Review by seanbudd — APPROVED** [review]

**Pull request merged**